project/src/main/webapp/document.jsp

```diff
@@ -172,7 +172,7 @@
         var a = document.createElement('a');
         var blob = new Blob([text], {'type':contentType});
         a.href = window.URL.createObjectURL(blob);
-        a.download = '<%= document.getName() %>' + "." + extDict["<%= document.getLanguage() %>"];
+        a.download = '<%= document.getName().replace("\'", "\\\'") %>' + "." + extDict["<%= document.getLanguage() %>"];
```

Comment:
can you handle this more extensively with other possible special characters like *, ? --> you can use regex to remove all non abc characters

Reply:
In practice, other characters do not cause issues. The problem that was happening was if there was a single quote, when it was injected into the scriptlet it would close the string and leave characters outside of it, causing the code to break. With this solution we can allow them to use whichever characters they would like